### ISSUE TITLE
An idea for (perhaps) a better naming of note clones.

### ISSUE THREAD

**Issue by XXXJetfireXXX** [zadam/trilium#4099]
### Describe feature

Like it's said in the wiki,
> I call these "clones", but that is a bit misleading - there's no original and cloned note - the notes in both of the parents categories are identical.

> Another misleading thing about "cloning" is that it suggests that a copy of the note has been made. That's not really true, the note itself stays in just one original, it is just referenced in multiple places in the tree hierarchy. So changing it in one category changes it in all the others, because they're all the same note.

Indeed, the name 'clones' can be somewhat misleading as it implies that there must be an 'original'. But there's not - note clones are more like hardlinks in Windows. Multiple instances of the same note that can appear in several places inside the note tree, but are all essentially the same note.
And I just thought: what if we call them `note instances` instead?

### Additional Information

In general, each time a program runs, it is an instance of that program. In OOP languages, when we create an object of a class, it is an instantiation of that class.
In similar way, as far as I can tell, a physical copy of a file is never called an instance, but a hardlink can.

I'm not sure if such naming would make things less or more complicated though, it's just a random idea that I got. Let me know what you think about it.

**Comment by zadam** [comment]
In my understanding, an "instance" implies the possibility of having different content. I can e.g. make two instances of class `ArrayList`, each having different elements. But "clones" actually have the same content.

**Comment by Dwindle2234** [comment]
Could we consider "mirror"?  I think it fits nicely.

**Comment by XXXJetfireXXX** [comment]
> In my understanding, an "instance" implies the possibility of having different content. I can e.g. make two instances of class ArrayList, each having different elements. But "clones" actually have the same content.
True... Now I understand why I still had a subtle feeling that it can also be misleading. That's a fair point x)
> Perhaps 'linked clone' would be better at the expense of a longer name for the feature.
> Could we consider "mirror"? I think it fits nicely.
I like these! While 'linked clone' is a more elaborative variant, 'mirror' also represents the true nature of the feature pretty precisely. However, imo when you want to refer to a particular cloned note, it would be more convenient to call it 'linked clone' rather than 'note mirror' or 'mirrored note' (note reflection? :D).

**Comment by XXXJetfireXXX** [comment]
'Linked clone' sounds like a good compromise between not changing the feature's name too much (less confusion for old users) while making its name less misleading in terms of what the feature does (less confusion for new users). It also goes as far as possible from the 'original/not original' concept, while keeping the word 'clones' intact.

**Comment by Nriver** [comment]
Notes in note tree are `references` to the original data. To end user, the cloned ones are `synchronized copies`.

But I'd go with `clone`, as it's shorter and easy to understand for new users.

**Comment by XXXJetfireXXX** [comment]
Sure, it is right that we don't actually see the data itself, we see the references that point to it within the note tree. But that level of abstraction (working with 'pointers' to data, not just notes) could only cause more confusion without understanding how data is being stored on a slightly deeper level x)

**Comment by Dwindle2234** [comment]
_"Notes in note tree are references to the original data"_

Would this make every note a 'clone'...

**Comment by XXXJetfireXXX** [comment]
It just means that what we see are pointers to note data, not the notes themselves. :)
There might be one 'pointer' to a note, or there might be multiple of these as well, all representing the same note in the tree.

**Comment by XXXJetfireXXX** [comment]
The closest analogy I can get are hard links, like mentioned above.

**Comment by Nriver** [comment]
> _"Notes in note tree are references to the original data"_
> 
> Would this make every note a 'clone'...

Well, technically, this is right. A note in the note tree is 1 clone of the data. If there is more than 1 clone, there will be a html class called "multiple-parents" apply to the note in the tree, which will show a "*" after the note.
https://github.com/zadam/trilium/blob/dc0a0dcf094f904bff4da3abc18f405470b38b37/src/public/app/widgets/note_tree.js#L771C17-L771C27

**Comment by shutaozhenzhen** [comment]
I think `linked clone` just more misleading, for it means there is a origin note and a link point to it.
`mirror` is better, but I like use `refer` for clone is doing reference

**Comment by salmund** [comment]
How about "mirror notes" ?

**Comment by rauenzi** [comment]
I like to think of every node in the tree as a pointer, so a clone note is just another pointer to the same note! But that's not very accessible terminology for less technical users. I think of all the suggestions so far, Mirror is perhaps the best option in terms of accuracy and accessibility to users.

Other terms I would throw out there would be `peer` (or `companion`) notes, `joint` (or `joined`) notes, `reciprocal` notes,  or if we want to stick to the family-style naming like child, parent, sibling, etc, we could call them cousins 😄

**Comment by sottey** [comment]
I completely agree with the use of "mirror". Clone works for me, but mirror seems more "correct". A true clone originates from the copied item, but then can deviate from the base (in biology, that is), so mirror feels right.

view could also be used, as it indicates a view into the original, but view has been used to describe a subset of a main repository, so that isn't quote right.

Thanks for getting me thinking on this!

**Comment by zadam** [comment]
Hmm, "mirror" sounds quite good indeed.

**Comment by DreamHoney-J** [comment]
How about "shadow".

**Comment by rom1dep** [comment]
I came up with "mirror"/"mirrored note" independently as part of #3087 (thanks @meichthys for the pointer),

I thought of "Shortcut" and "Link" as well, but the common acceptance for those two is that they derive from an "original" that is "truer" (as in, removing the "source" breaks the "reference"), whereas "mirrors" or "clones" are supposedly equivalent and impossible to tell apart, which I think is overall a more accurate characterisation of how trilium is handling this.

Hence my vote goes to "mirror"

**Comment by youli42** [comment]
为什么不试试硬链接？这大概是一个文件系统中的概念，但他的作用和这里的克隆几乎一致：将一个文件放在多个地方，只占用一份空间，在修改中保持一致，只有在删除所有链接时才会真正删除文件。

希望翻译工具可以正确翻译这些术语

Why not try hard links? This is probably a concept in a file system, but its function is almost the same as cloning here: placing a file in multiple locations, occupying only one copy of space, maintaining consistency during modifications, and the file will only be truly deleted when all links are removed.

Hope the translation tool can correctly translate these terms.

**Comment by eliandoran** [comment]
@youli42 , hard link is technically correct, but it's not a very user friendly name. Only power users (and probably even then it would be mostly familiar to Linux/Unix users rather than Windows) would know about it so I would prefer we avoid very technical terms.

**Comment by youli42** [comment]
理解您的顾虑，感谢您愿意为了新手着想。

但我认为，无论使用哪个词，一般用户使用时，同样需要阅读说明才能理解和使用；而使用技术准确的定时，可以让部分用户直观的理解笔记功能。

综上所述，使用稍微专业一点的词汇是不是利大于弊？

如果不认同，不必回复，我也不会再纠结这个问题，你们的工作一直非常棒！感谢您和各位开发者对开源社区的贡献！

I understand your concerns and appreciate that you are thinking of new users.

However, I believe that no matter which term is used, ordinary users will still need to read the instructions to understand and use it; while using the technically accurate term "timing" can allow some users to intuitively understand the note function.

To sum up, is using slightly more professional vocabulary more beneficial than harmful?

If you disagree, there's no need to reply. I won't dwell on this issue anymore. Your work has always been excellent! Thank you and all the developers for your contributions to the open-source community!

**Comment by itsjakobx** [comment]
I think the problem isn’t with the verb “cloning,” which we’ve simply projected our own meaning onto, but with the fact that we never defined what thing (a noun) it actually creates.

If you clone a note it becomes some thing. Lets call that thing X.

-> cloning a note creates X.

cloning _automatically implies_ creation of clones, i.e. X = clones. One expectes that cloning creates clones. But in the noun "clone" we already have a prejudice view about its semantics. Since Verbes define some change of a subject (in this case notes) towards an object (X) we can freely change its meaning without confusing the user, by giving the object X a reasonable name – Even though English doesn’t have compound words (like in German) it has compound-like noun phrases still sounding natural – Example: data link, hard link, synced peer, ...— which gives us more freedom when naming things.

To remove the implication we can explicitly tell the user the noun X that the verb clone creates. So we would have smth like: X = synced peers OR X = linked peers OR X = hard links etc, etc.
The result could look like:
-> cloning a note creates synced peers 

Thus the problem is solved as the correct meaning of cloning becomes obvious. Its then homonyms nature becomes irrelevant.

>I call these “clones”, but that is a bit misleading - there’s no original and cloned note - the notes in both of the parents categories are identical [^1].

is just self aware, because it recognizes that "to clone" imples creating "clones" but couldn't solve the resulting ambiguity, it thus recognized defeat for that time being. 

To sum up: we still missed out on defining what cloning a note creates and this is what determines the meaning of cloning itself. Verbs don't matter just pick a good sounding one, and choose the interacting noun wisely.

[^1]: https://triliumnext.github.io/Docs/Wiki/cloning-notes.html

**Comment by contributor** [comment]
coplace?

**Comment by eliandoran** [comment]
> coplace

Might be difficult for non-native speakers since the "co-" prefix and "place" doesn't immediately ring a bell. On my side I still prefer the "mirror" terminology.

**Comment by meichthys** [comment]
I do think `mirror` sufficiently fits the bill. It would be succinct and understandable by new users and can act as both a noun and a verb. i.e:

as a noun:
`Create a mirror of this note`, `This is a mirror of another note`, `This note has 4 mirrors`
or as a verb:
`Mirror this note`, `This note is mirrored in 4 places`, etc

**Comment by itsjakobx** [comment]
Looking at the thread, I dont see any comment disagreeing with "mirror". Are we fighting noone?

**Comment by eliandoran** [comment]
@werererer , absolutely. Regardless of the terminology, there's a lot of work to do to "rebrand" this functionality.

**Comment by XXXJetfireXXX** [comment]
Tbh I'm surprised that this thread is still alive to this day xD
